src/tests/tracing/eventpipe/allocationsampling/allocationsampling.cs

```diff
@@ -0,0 +1,255 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers.Clr;
+using Microsoft.Diagnostics.NETCore.Client;
+using Tracing.Tests.Common;
+using Xunit;
+
+namespace Tracing.Tests.SimpleRuntimeEventValidation
+{
+    public class AllocationSamplingValidation
+    {
+        // variables used to compare results of different tests
+        static long _noSamplingDuration = 0;
+        static long _allocationTickDuration = 0;
+        static long _allocationSampledDuration = 0;
+
+        [Fact]
+        public static int TestEntryPoint()
+        {
+            // check that AllocationSampled events are generated and size + type name are correct
+            var ret = IpcTraceTest.RunAndValidateEventCounts(
+                new Dictionary<string, ExpectedEventCount>() { { "Microsoft-Windows-DotNETRuntime", -1 } },
+                _eventGeneratingActionForAllocations,
+                // AllocationSamplingKeyword (0x80000000000): 0b1000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000
+                new List<EventPipeProvider>() { new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Informational, 0x80000000000) },
+                1024, _DoesTraceContainEnoughAllocationSampledEvents, enableRundownProvider: false);
+            if (ret != 100)
+                return ret;
+
+            // measure impact of AllocationTick and AllocationSampled
+            ret = RunAllocationSamplers();
+            if (ret != 100)
+                return ret;
+
+            return 100;
+        }
+
+        private static int RunAllocationSamplers()
+        {
+            //// trigger GC to avoid one during the next tests
+            //GC.Collect();
+
+            // run the same allocations test with no sampling, AllocationTick and AllocationSampled
+            Stopwatch clock = new Stopwatch();
```

Comment:
If you were just poking around this is fine, but prior to the PR getting checked in this performance testing shouldn't be enabled here. There are some allocation perf tests in the perf repo that I'd expect to detect regressions:
https://github.com/dotnet/performance/tree/main/src/benchmarks/micro/runtime/GC
https://github.com/dotnet/performance/tree/main/src/benchmarks/gc

We don't typically have automated perf tests that measure overhead when events are turned on, instead we just test it manually when doing the work to add the events or making substantial changes.

Reply:
I did not know where to put these "tests" that are not validating any expected result but simply provide an hint about the perf differences between no sampling, AllocationTick and the new one that you could see. This is what I get on my machine:

I'm not sure it would help detecting regression so I will remove them soon

Reply:
I think you could put the code for manual tests in a separate file in this directory + a README saying it is only for manual testing. I don't think the code needs to build or run in an automated way, manually invoking dotnet build seems fine. Probably nobody else will run it in the future but it would be nice for me and other reviewers to be able to see exactly what was validated.

Reply:
I created a dedicated folder with README.md